api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -495,17 +492,41 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
 
   @Override
   public ResponseEntity<WorkspaceResponseListResponse> getWorkspaces() {
-    // TODO: use FireCloud to determine what workspaces to return, instead of just returning
-    // workspaces from our database.
     User user = userProvider.get();
+    // This code is to migrate firecloud uuid to current workspaces
+    // once we are confident this has been done we can rework/remove
+    // much of this logic
     List<WorkspaceResponse> responseList = new ArrayList<WorkspaceResponse>();
     if (user != null) {
+      List<org.pmiops.workbench.firecloud.model.WorkspaceResponse> fcWorkspaces;
+      try {
+        fcWorkspaces = fireCloudService.getWorkspaces();
+      } catch (org.pmiops.workbench.firecloud.ApiException e) {
+        throw ExceptionUtils.convertFirecloudException(e);
+      }
       for (WorkspaceUserRole userRole : user.getWorkspaceUserRoles()) {
-        // TODO: Use FireCloud to determine access roles, not our DB
-        WorkspaceResponse currentWorkspace = new WorkspaceResponse();
-        currentWorkspace.setWorkspace(TO_CLIENT_WORKSPACE.apply(userRole.getWorkspace()));
-        currentWorkspace.setAccessLevel(userRole.getRole());
-        responseList.add(currentWorkspace);
+        org.pmiops.workbench.firecloud.model.WorkspaceResponse fcWorkspace;
+        org.pmiops.workbench.db.model.Workspace dbWorkspace;
+        dbWorkspace = workspaceService.getRequired(userRole.getWorkspace().getWorkspaceNamespace(), userRole.getWorkspace().getFirecloudName());
```

Comment:
Why isn't this just userRole.getWorkspace()?

This is going to cause O(n) DB lookups...

Reply:
Was there a plan for follow-up here?